test/test_container.py

```diff
@@ -0,0 +1,87 @@
+import os
+import subprocess
+import sys
+import argparse
+import subprocess
+
+def test_singularity_container(model_name, candle_data_dir, gpuid, definitions_file, deployment_dir_file, options="--epochs 1  "):
+    """
+    Build singularity container from model_name and singularity_dir, and run it for 1 epoch
+    """
+
+    if os.path.isfile(deployment_dir_file):
+        print("Singularity container exists")
+    else:
+        print("Singularity container does not exist", deployment_dir_file)
+        # exit(1)
+
+    print("Running singularity container")
+    
+    cmd = "singularity run --nv --bind " + candle_data_dir + ":/candle_data_dir " + deployment_dir_file + " train.sh " + gpuid + " /candle_data_dir " + options
+    
+    print("Running command:", cmd)
+    
+    process = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    stdout, stderr = process.communicate()
+
+    print("Standard Output:")
+    # print(stdout.encode("utf-8").decode("utf-8"))
+    print(stdout)
+
+    print("\nStandard Error:")
+    # print(stderr.encode("utf-8").decode("utf-8"))
+    print(stderr)
+    
+    return process.returncode
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Build Singularity container. Launch from Singularity/test as Eg. 1: python build_container.py --model_name DeepTTC  .OR. Eg.2: python build_container.py --model_name GraphDRP --definitions_file /PathTo/GraphDRP.def --deployment_dir_file /PathTo/GraphDRP.sif")
+    parser.add_argument("--model_name", help="Name of the model", default="GraphDRP")
+    parser.add_argument("--candle_data_dir", help="Path to the candle data directory", default="/tmp/")
```

Comment:
Might be better to change the default to "tmp/". When I run `python test_container.py --model_name tCNNS`, I get 
```NotFoundError: [Errno 2] No such file or directory: \'/candle_data_dir/tCNNS/Data/data_processed/drug_onehot_smiles.npy```

Reply:
This command worked: `python test_container.py --model_name tCNNS --definitions_file ../definitions/tCNNS.def --candle_data_dir tmp/`

Reply:
My tmp folder is in /homes/ac.sejones/test/Singularity/test

Reply:
Fixed.